pre_commit/hook_executor.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import unicode_literals
```

Comment:
so, my main issue with this is the `Repository` is supposed to be the hook runner, now we'd have two things that have that responsibility...

I think we should maybe instead push the responsibility into `Repository.create`?